src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/DefaultRazorConfigurationService.cs

```diff
@@ -84,25 +74,48 @@ public async override Task<RazorLSPOptions> GetLatestOptionsAsync(CancellationTo
             }
         }
 
-        private RazorLSPOptions BuildOptions(IConfiguration config)
+        private static RazorLSPOptions BuildOptions(JObject[] result)
```

Comment:
This logic is important to test. It's worth making this internal and adding a few tests to ensure that we generate the correct lsp options